isoimage.py

```diff
@@ -0,0 +1,151 @@
+        mkdir_p(target_directory)
+
+        #get_iso_package_name(data_loaded,packages);
+        image_name = 'default.iso'
+        image_type = 'iso'
+        structure = {}
+        for row in data_loaded.get('packages'):
+            if 'structure' in row:
+                image_name = row.get('name')
+                image_type = row.get('type')
+                print('Image name:', image_name)
+                print('Image type:', image_type)
+                structure  = row.get('structure')
+                for files_row in structure.get('files'):
+                    if 'file' in files_row:
+                        file_name = files_row.get('file')
+                        unarhive = files_row.get('unarchive')
+                        py = pathlib.Path().glob(file_name)
+                        for file in py:
+                            shutil.copy2(file, target_directory)
+                for subdirs_row in structure.get('subdirs'):
+                    process_subdir(subdirs_row, None)
+        command_name += '  -V CDNAME -J -r -o '+image_name+' '+target_directory;
+        print(" Generating Iso image", command_name)
+        os.system(command_name)
+
+def isNotValid(file_name):
+    print("ERROR: File:"+file_name+'is not Valid')
+
+def create_destination_directory(parent_dir, subdir):
+    target_directory = 'target';
+    #print('Subdirectory:',subdir)
+    #print('Parent:',parent_dir)
+    if(parent_dir is not None):
+       destination = get_destination_directory_name(target_directory,parent_dir)
+       destination = get_destination_directory_name(destination,subdir)
+    else:
+        destination = get_destination_directory_name(target_directory,subdir)
+        
+    mkdir_p(destination)
+    return destination;
+    
+
+
+def process_subdir(subdirs_row, parent_dir):
+    subdir = subdirs_row.get('subdir')
+    #print('Subdirectory1:',subdir)
+    #print('Parent1:',parent_dir)
+    destination = create_destination_directory(parent_dir, subdir)
+    #print('destination1:',destination)
+    if "subdirs" in subdirs_row:
+        if( subdirs_row.get('subdirs') is not None) :
+            for sub_subdirs_row in subdirs_row.get('subdirs'):
+                #print("Processing subdirs:"+sub_subdirs_row.get('subdir'))
+                process_subdir(sub_subdirs_row,subdir)
+
+    if 'files' in subdirs_row:
+        for files_row in subdirs_row.get('files'):
+            file_name = files_row.get('file')
+            unarhive = files_row.get('unarchive')
+            #print('Processsing file_name:',file_name)
+            #print('should Unarchive:',unarhive)
+            if unarhive :
+               unarhive_file_to_destination(destination, file_name)
+            else:
+                #print('Copy file to destination')
+                py = pathlib.Path().glob(file_name)
+                for file in py:
+                    shutil.copy2(file, destination)
+
+def unarhive_file_to_destination(destination, file_name):
```

Comment:
Should be unarchive_file_to_destination. This will have to be fixed in other places too

Reply:
fixed in this push 
97c1590